qiskit/circuit/library/n_local/n_local.py

```diff
@@ -934,8 +942,15 @@ def _build(self) -> None:
             return
 
         # use the initial state circuit if it is not None
+        # if the initail_state is an instance of QuantumCircuit, do not call construct_circuit
         if self._initial_state:
-            circuit = self._initial_state.construct_circuit("circuit", register=self.qregs[0])
+            if isinstance(self._initial_state, QuantumCircuit):
+                circuit = self._initial_state.copy()
+            else:
+                warn_message = """'InitialState' is now deprecated and will be removed soon.
```

Comment:
Could you use the standard deprecation warning we use in Qiskit: https://qiskit.org/documentation/contributing_to_qiskit.html#deprecation-warnings 🙂